uuid.go

```diff
@@ -153,6 +156,28 @@ func (u *UUID) SetVariant(v byte) {
 	}
 }
 
+// UnixNanoOf returns the number of nanoseconds in a V1 UUID
+func (u *UUID) UnixNanoOf() (int64, error) {
+	if u.Version() != 1 {
+		err := fmt.Errorf("uuid: %s is version %d, not version 1", u, u.Version())
+		return 0, err
+	}
+	low := binary.BigEndian.Uint32(u[0:4])
+	mid := binary.BigEndian.Uint16(u[4:6])
+	hi := binary.BigEndian.Uint16(u[6:8]) & 0xfff
+	return 100 * (int64(low) + (int64(mid) << 32) + (int64(hi) << 48) - int64(epochStart)), nil
+}
+
+// TimeOf returns the time embedded in a V1 UUID
+// An error is returned if this is not a V1 UUID
+func (u *UUID) TimeOf() (time.Time, error) {
```

Comment:
Would you mind converting this to a function on the package that takes the UUID as input? Because this method only takes a UUID, I think it's more appropriate to be a function. If the UUID wasn't the input to the method, but was the state needed for the method to take action on the input, then I'd say 👍 on it being a method.